charmhelpers/contrib/hahelpers/cluster.py

```diff
@@ -404,11 +404,25 @@ def distributed_wait(modulo=None, wait=None, operation_name='operation'):
         # will still wait
         calculated_wait = modulo_distribution(modulo=modulo, wait=wait,
                                               non_zero_wait=True)
-    msg = "Waiting {} seconds for {} ...".format(calculated_wait,
-                                                 operation_name)
-    log(msg, DEBUG)
-    status_set('maintenance', msg)
-    time.sleep(calculated_wait)
+
+    # update status every 60 seconds with wait time elapsed
+    remaining_wait = calculated_wait
+    elapsed = 0
+    while remaining_wait:
```

Comment:
I think this would be more robust if it said:

```python
while remaining_wait > 0:
    ...
```

I know that by inspection the code will work, it's just the defensive programmer in me!